src/app/core/common-components/entity-form/unique-id-validator/unique-id.service.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from "@angular/core";
+import { EntityMapperService } from "../../../entity/entity-mapper/entity-mapper.service";
+import { uniqueIdValidator } from "./unique-id-validator";
+
+@Injectable({ providedIn: "root" })
+export class UniqueIdService {
```

Comment:
Did you have a look at [async validators](https://angular.io/api/forms/AsyncValidator)? As far as I understood it, they would allow to move the async part to the validation check instead of where the validator is built.